mozdownload/scraper.py

```diff
@@ -242,6 +243,51 @@ def detect_platform(self):
         else:
             return "%s%d" % (mozinfo.os, mozinfo.bits)
 
+    def get_checksum_url(self):
+        """Get the checksum url for current version."""
+        filename = 'SHA512SUMS'
+        url = os.path.split(os.path.split(os.path.split(self.url)[0])[0])[0] + '/' + filename
+        return url
+
+    def has_checksum(self, url):
+        """Check if the specified checksum file exists."""
+        self.logger.info('Checking checksum at url: %s' % url)
+        r = self.session.get(url, stream=True)
+        code = r.status_code
+        return code == 200
+
+    def download_checksum(self):
+        """Download the checksum."""
+        r = self.session.get(self.curl, stream=True)
+        r.raise_for_status()
+
+        with open(self.cfilename, 'wb') as f:
+                for chunk in r.iter_content(CHUNK_SIZE):
+                    f.write(chunk)
+
+    def get_checksum_name(self):
+        """Get the name of the platform in the checksum file."""
+        path = (os.path.split(os.path.split(os.path.split(self.url)[0])[0])[1] +
+                '/' + os.path.split(os.path.split(self.url)[0])[1] + '/' +
+                os.path.split(self.url)[1])
+        return path
```

Comment:
Please use the available regex patterns we already have in use for various scraper classes. Basically it should work similarly as when running the following code: `mozdownload -t daily --extension checksums`. So I don't think that we have to reinvent the wheel, and can base on the already existing logic.